### ISSUE TITLE
Saltcorn not responding after disk reached max storage

### ISSUE THREAD

**Issue by reclawe** [saltcorn/saltcorn#1699]
Our saltcorn installation stopped responding to HTTP requests and after SSHing in we found it's at 100% storage. Is there anyway to make a backup through SSH (downloading the backup zip via SSH)? We'd like to at least recover our data stored on the machine.

**Comment by glutamate** [comment]
1. make sure you are the saltcorn user and cd to their home directory
2. run `saltcorn backup -z` 

it print the name of the backup file

**Comment by reclawe** [comment]
Running this brings up this error:

`Error reading mjmlconfig :  Error: EACCES: permission denied, open '/root/.mjmlc                                                                                                             onfig'
    at Object.openSync (fs.js:476:3)
    at Object.readFileSync (fs.js:377:35)
    at readMjmlConfig (/home/saltcorn/.local/lib/node_modules/@saltcorn/cli/node                                                                                                             _modules/mjml-core/lib/helpers/mjmlconfig.js:38:43)
    at handleMjmlConfig (/home/saltcorn/.local/lib/node_modules/@saltcorn/cli/no                                                                                                             de_modules/mjml-core/lib/helpers/mjmlconfig.js:146:7)
    at Object.<anonymous> (/home/saltcorn/.local/lib/node_modules/@saltcorn/cli/                                                                                                             node_modules/mjml-core/lib/index.js:495:27)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/home/saltcorn/.local/lib/node_modules/@saltcorn/cli/                                                                                                             node_modules/mjml/lib/index.js:12:41)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14) {
  errno: -13,
  syscall: 'open',
  code: 'EACCES',
  path: '/root/.mjmlconfig'
}
[Error: EACCES: permission denied, mkdir '/root/plugin_packages'] {
  errno: -13,
  code: 'EACCES',
  syscall: 'mkdir',
  path: '/root/plugin_packages'
}
...
    Error: EACCES: permission denied, mkdir './.'
    Code: EACCES
`
We recently upgraded the disk storage but saltcorn is still down.

**Comment by glutamate** [comment]
Looks like you are running this from the `/root` directory. Make sure you switch to the saltcorn user (with `su`) and then cd to their home directory (just run `cd`)

**Comment by reclawe** [comment]
Thanks for that, I was able to get that backup downloaded. Our server is still down however, connection attempt gives me Connection Refused error in the browser. Would you know what I could check or reconfigure to get it up and running again?

**Comment by nomesaili** [comment]
Hi glutamate. We tried while it does generate a backup zip. Some of the key tables are missing when we restore to a new server. 

This is the output screen when running saltcorn backup -z command. Note that we already run apt-upgrade as well. 

Any other actions we can take?

Any help apperciated

error: permission denied to create extension "uuid-ossp"
    at Parser.parseErrorMessage (/home/saltcorn/.local/lib/node_modules/@saltcorn/cli/node_modules/pg-protocol/src/parser.ts:369:69)
    at Parser.handlePacket (/home/saltcorn/.local/lib/node_modules/@saltcorn/cli/node_modules/pg-protocol/src/parser.ts:188:21)
    at Parser.parse (/home/saltcorn/.local/lib/node_modules/@saltcorn/cli/node_modules/pg-protocol/src/parser.ts:103:30)
    at Socket.<anonymous> (/home/saltcorn/.local/lib/node_modules/@saltcorn/cli/node_modules/pg-protocol/src/index.ts:7:48)
    at Socket.emit (events.js:400:28)
    at addChunk (internal/streams/readable.js:293:12)
    at readableAddChunk (internal/streams/readable.js:267:9)
    at Socket.Readable.push (internal/streams/readable.js:206:10)
    at TCP.onStreamRead (internal/stream_base_commons.js:188:23) {
  length: 166,
  severity: 'ERROR',
  code: '42501',
  detail: undefined,
  hint: 'Must be superuser to create this extension.',
  position: undefined,
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'extension.c',
  line: '806',
  routine: 'execute_extension_script'

**Comment by glutamate** [comment]
How are you trying to restore? The best way of doing that is with a brand-new installation on the create first user  screen, where there is a restore option

**Comment by nomesaili** [comment]
Yes we tried that with a new installtion of saltcorn. We are using an the saltorn image on digital ocean marketplace

**Comment by glutamate** [comment]
How do you see that error? is it in the crash log? Can you post some screenshots?

**Comment by nomesaili** [comment]
Hi glutamate

See attach screen from terminal once I run the backup

![SC](https://user-images.githubusercontent.com/49111643/224451087-14b8164f-ddbf-457e-9f0d-7d97bfde28ed.png)

Thats cool with the disk usage indicator. Its our fault though. We didn't limit the file size upload why I believe the disk was used up to fast

**Comment by reclawe** [comment]
Hi glutamate,

Is there a way we can manually extract the table data through SSH or other methods without using the zip? And where can I find the crash logs you mentioned?

**Comment by reclawe** [comment]
Thanks for that glutamate. At least we have that data now but about restoring it...

When trying to restore using backup zip mentioned previously, it comes up with the error in the image attached:

![image](https://user-images.githubusercontent.com/61644025/225460235-514f6064-8e1b-4a69-8a79-6044e5f4858e.png)

I know this has to do with the ID value being unsynced, would you know the best way to fix this? If we can just fix the process of restoring using the backup zip that should enable us to finally continue operations.

**Comment by glutamate** [comment]
Are you sure you are not trying to restore twice? Or that for some reason there is some data remaining on the instance you are restoring on? It should be completely blank and idea to create first user screen

**Comment by reclawe** [comment]
Leaving this comment here for future users:

I managed to figure out a couple of things. Unsure why saltcorn shutdown in the first place but the saltcorn.service got stuck in a restart loop which eventually maxed out the storage to our server via constant writing to the log files found in var/log.

Trying to start it up again brought up the permission issue - basically the startup service got messed up somehow and needed to be reconfigured so it starts correctly.

To fix this I ran through the installation process again described on the saltcorn wiki. Backups were all fine.

Thanks glutamate
